proposals/lip-0043.md

```diff
@@ -197,85 +164,164 @@ sidechainRegParams = {
+        "certificateThreshold": trs.params.certificateThreshold
+    }
+    create an entry in the chain validators data substore with
+        storeKey = chainID
+        storeValue = encode(validatorsSchema, sidechainValidators)
+
+    # Create outbox root entry.
+    create an entry in the outbox root substore with
+        storeKey = chainID
+        storeValue = encode(outboxRootSchema, {"root": sidechainChannel.outbox.root})
+
+    # Create registered names entry.
+    create an entry in the registered names substore with
+        storeKey = trs.params.name
+        storeValue = encode(chainIDSchema, {"ID": chainID})
+
+    # Burn the registration fee.
+    Token.burn(senderAddress, TOKEN_ID_LSK_MAINCHAIN, REGISTRATION_FEE)
+
+    # Emit chain account updated event.
+    emitEvent(
+        module = MODULE_NAME_INTEROPERABILITY,
+        name = EVENT_NAME_CHAIN_ACCOUNT_UPDATED,
+        data = sidechainAccount,
+        topics = [chainID]
+    )
+
+    # Send registration CCM to the sidechain.
     registrationCCMParams = {
-        "networkID": SHA-256(trs.params.genesisBlockId || senderAddress),
+        "chainID": chainID,
         "name": trs.params.name,
         "messageFeeTokenID": sidechainChannel.messageFeeTokenID
     }
-    ```
-
-* An entry in the chain validators substore as specified in [LIP 0045][lip-0045#chain-substore] is created as:
-  * `storePrefix`: `STORE_PREFIX_CHAIN_VALIDATORS`
-  * `storeKey`: `uint32be(chainID)`.
-  * `storeValue`: `sidechainValidators` where `sidechainValidators` is the validators object, serialized according to the `validatorsSchema` schema as defined in [LIP 0045][lip-0045], and initialized as:
-    * The `sidechainValidators.activeValidators` property is equal to the `trs.params.initValidators` property.
-    * The `sidechainValidators.certificateThreshold` property is equal to the `trs.params.certificateThreshold` property.
-* An entry in the outbox root substore as specified in [LIP 0045][lip-0045] is created as:
-  * `storePrefix`: `STORE_PREFIX_OUTBOX_ROOT`.
-  * `storeKey`: `uint32be(chainID)`.
-  * `storeValue`: The property `sidechainChannel.outbox.root` serialized according to the `outboxRootSchema` schema defined in Introduce Interoperability module LIP.
-* An entry in the registered names substore as specified in [LIP 0045][lip-0045] is created as:
-  * `storePrefix`: `STORE_PREFIX_REGISTERED_NAMES`.
-  * `storeKey`: `sidechainAccount.name` serialized as a utf-8 encoded string.
-  * `storeValue`: `chainID `serialized according to `chainIDSchema` schema defined in Introduce Interoperability module LIP.
-* An entry in the registered network IDs substore as specified in [LIP 0045][lip-0045] is created as:
-  * `storePrefix`: `STORE_PREFIX_NETWORK_ID`.
-  * `storeKey`: `sidechainAccount.networkID `serialized as bytes.
-  * `storeValue`: `chainID` serialized according to `chainIDSchema` schema as defined in [LIP 0045][lip-0045].
+    send(EMPTY_FEE_ADDRESS, 
```

Comment:
Will this message be sent? I think, it should raise an exception because the status of the chain with ID `chainID` will be `CHAIN_STATUS_REGISTERED`, not `CHAIN_STATUS_ACTIVE`. Am I missing something?

Reply:
nice catch, I will explicitly avoid the check for this first message

Reply:
Looks good now, I just still have one question.
Where do we update the status from `CHAIN_STATUS_REGISTERED` to `CHAIN_STATUS_ACTIVE`?

Reply:
registration ccm execution in LIP49